ionc/ion_reader.c

```diff
@@ -1917,7 +1917,12 @@ iERR _ion_reader_process_possible_symbol_table(ION_READER *preader, BOOL *is_sym
         preader->_local_symtab_pool = owner;
         preader->_current_symtab = local;
     }
-    iRETURN;
+    SUCCEED();
```

Comment:
This is how SUCCEED is defined:
```
#define SUCCEED()        { err = IERR_OK; goto fail; }
```
So this line will jump to the `fail` label on the next line and free the owner, if any. It looks like that's not actually what we want; in the successful case we don't want to free the owner. This line should be replaced by `return IERR_OK;`

I think this is why the tests are failing.

Reply:
Got it thanks. Change it to `return IERR_OK` works.